Using paths inside tsconfig causing error on JIT: field 'browser' doesn't contain a valid alias configuration

### Bug Report or Feature Request (mark with an `x`)
```
- [x] bug report -> please search issues before submitting
- [ ] feature request
```
While using the `paths` object inside `tsconfig.json`, module resolution doesn't seem to work anymore since the latest release. It did work on `1.3.0-rc.5`. Ahead of Time does work. This seems to happen when you reference the `src` folder or anything lower down the tree

### Versions.
@angular/cli: 1.3.0
node: 8.2.1
os: win32 x64
@angular/animations: 4.3.3
@angular/common: 4.3.3
@angular/compiler: 4.3.3
@angular/core: 4.3.3
@angular/forms: 4.3.3
@angular/http: 4.3.3
@angular/platform-browser: 4.3.3
@angular/platform-browser-dynamic: 4.3.3
@angular/router: 4.3.3
@angular/cli: 1.3.0
@angular/compiler-cli: 4.3.3
@angular/language-service: 4.3.3

### Repro steps.
0. Upgrade cli to version `1.3.0`, this used to work in `1.3.0-rc.5` and before
1. Create a project
2. Set a paths property to `"@root/*": ["./*"]` inside `tsconfig.app..json`
3. Change the import of `AppComponent` inside `AppModule` to `import { AppComponent } from '@root/app/app.component';`
4. Run `ng serve`
5. Run `ng serve --aot` to see that this does work

### The log given by the failure.
```
ERROR in ./src/app/app.module.ts
Module not found: Error: Can't resolve '@root/app/app.component' in 'C:\Projects\angular-cli-v130-bug\src\app'
resolve '@root/app/app.component' in 'C:\Projects\angular-cli-v130-bug\src\app'
  Parsed request is a module
  using description file: C:\Projects\angular-cli-v130-bug\package.json (relative path: ./src/app)
    Field 'browser' doesn't contain a valid alias configuration
  after using description file: C:\Projects\angular-cli-v130-bug\package.json (relative path: ./src/app)
    resolve as module
      C:\Projects\angular-cli-v130-bug\src\app\node_modules doesn't exist or is not a directory
      C:\Projects\angular-cli-v130-bug\src\node_modules doesn't exist or is not a directory
      C:\Projects\node_modules doesn't exist or is not a directory
      C:\node_modules doesn't exist or is not a directory
      looking for modules in C:\Projects\angular-cli-v130-bug\node_modules
        using description file: C:\Projects\angular-cli-v130-bug\package.json (relative path: ./node_modules)
          Field 'browser' doesn't contain a valid alias configuration
        after using description file: C:\Projects\angular-cli-v130-bug\package.json (relative path: ./node_modules)
          using description file: C:\Projects\angular-cli-v130-bug\package.json (relative path: ./node_modules/@root/app/app.component)
            no extension
              Field 'browser' doesn't contain a valid alias configuration
              C:\Projects\angular-cli-v130-bug\node_modules\@root\app\app.component doesn't exist
            .ts
              Field 'browser' doesn't contain a valid alias configuration
              C:\Projects\angular-cli-v130-bug\node_modules\@root\app\app.component.ts doesn't exist
            .js
              Field 'browser' doesn't contain a valid alias configuration
              C:\Projects\angular-cli-v130-bug\node_modules\@root\app\app.component.js doesn't exist
            as directory
              C:\Projects\angular-cli-v130-bug\node_modules\@root\app\app.component doesn't exist
```

### Desired functionality.
I would like to see no errors, obviously ;)

### Mention any other details that might be useful.
Created a [repo](https://github.com/PierreDuc/angular-cli-v130-bug) showing this issue

Having the same issue with version 1.3.1: 
`ERROR in ./src/main/web/app/app.module.ts`
`Module not found: Error: Can't resolve '@module/api'`

and 

`Field 'browser' doesn't contain a valid alias configuration`

where `@module/api` ist defined in `tsconfig.app.json`

--- a/packages/@ngtools/webpack/src/paths-plugin.ts
+++ b/packages/@ngtools/webpack/src/paths-plugin.ts
@@ -122,29 +122,37 @@ export class PathsPlugin implements Tapable {
 
     this._nmf.plugin('before-resolve', (request: NormalModuleFactoryRequest,
                                         callback: Callback<any>) => {
+      // Only work on TypeScript issuers.
+      if (!request.contextInfo.issuer || !request.contextInfo.issuer.endsWith('.ts')) {
+        return callback(null, request);
+      }
+
       for (let mapping of this._mappings) {
         const match = request.request.match(mapping.aliasPattern);
         if (!match) { continue; }
-
         let newRequestStr = mapping.target;
         if (!mapping.onlyModule) {
           newRequestStr = newRequestStr.replace('*', match[1]);
         }
+        const moduleResolver = ts.resolveModuleName(
+          request.request,
+          request.contextInfo.issuer,
+          this._compilerOptions,
+          this._host
+        );
+        let moduleFilePath = moduleResolver.resolvedModule
+                          && moduleResolver.resolvedModule.resolvedFileName;
 
-        const moduleResolver: ts.ResolvedModuleWithFailedLookupLocations =
-          ts.nodeModuleNameResolver(
-            newRequestStr,
-            this._absoluteBaseUrl,
-            this._compilerOptions,
-            this._host
-          );
-        const moduleFilePath = moduleResolver.resolvedModule ?
-          moduleResolver.resolvedModule.resolvedFileName : '';
-
+        // If TypeScript gives us a .d.ts it's probably a node module and we need to let webpack
+        // do the resolution.
+        if (moduleFilePath && moduleFilePath.endsWith('.d.ts')) {
+          moduleFilePath = moduleFilePath.replace(/\.d\.ts$/, '.js');
+          if (!this._host.fileExists(moduleFilePath)) {
+            continue;
+          }
+        }
         if (moduleFilePath) {
-          return callback(null, Object.assign({}, request, {
-            request: moduleFilePath.includes('.d.ts') ? newRequestStr : moduleFilePath
-          }));
+          return callback(null, Object.assign({}, request, { request: moduleFilePath }));
         }
       }
 

--- a/tests/e2e/tests/build/ts-paths.ts
+++ b/tests/e2e/tests/build/ts-paths.ts
@@ -1,12 +1,12 @@
 import {updateTsConfig} from '../../utils/project';
-import {writeMultipleFiles, appendToFile, createDir} from '../../utils/fs';
+import {writeMultipleFiles, appendToFile, createDir, replaceInFile} from '../../utils/fs';
 import {ng} from '../../utils/process';
 import {stripIndents} from 'common-tags';
 
 
 export default function() {
   return updateTsConfig(json => {
-    json['compilerOptions']['baseUrl'] = '.';
+    json['compilerOptions']['baseUrl'] = './';
     json['compilerOptions']['paths'] = {
       '@shared': [
         'app/shared'
@@ -14,9 +14,8 @@ export default function() {
       '@shared/*': [
         'app/shared/*'
       ],
-      '*': [
-        '*',
-        'app/shared/*'
+      '@root/*': [
+        './*'
       ]
     };
   })
@@ -24,7 +23,15 @@ export default function() {
   .then(() => writeMultipleFiles({
     'src/meaning-too.ts': 'export var meaning = 42;',
     'src/app/shared/meaning.ts': 'export var meaning = 42;',
-    'src/app/shared/index.ts': `export * from './meaning'`
+    'src/app/shared/index.ts': `export * from './meaning'`,
+  }))
+  .then(() => replaceInFile('src/app/app.module.ts', './app.component', '@root/app/app.component'))
+  .then(() => ng('build'))
+  .then(() => updateTsConfig(json => {
+    json['compilerOptions']['paths']['*'] = [
+      '*',
+      'app/shared/*'
+    ];
   }))
   .then(() => appendToFile('src/app/app.component.ts', stripIndents`
     import { meaning } from 'app/shared/meaning';

